cadence/transactions/add_to_nft_catalog.cdc

```diff
@@ -11,29 +12,25 @@ transaction(
     nftID: UInt64,
     publicPathIdentifier: String
 ) {
-    let adminProxyResource : &NFTCatalogAdmin.AdminProxy
+    let adminProxyRef : &NFTCatalogAdmin.AdminProxy
 
-    prepare(acct: AuthAccount) {
-        self.adminProxyResource = acct.borrow<&NFTCatalogAdmin.AdminProxy>(from : NFTCatalogAdmin.AdminProxyStoragePath)!
+    prepare(acct: auth(BorrowValue, IssueStorageCapabilityController, PublishCapability, SaveValue, UnpublishCapability) &Account) {
```

Comment:
Same comment here about Account entitlements. It looks like all you are doing here is borrowing a reference from storage, so the only entitlement you need is `BorrowValue` I believe. You should go through all the transactions and make sure you are using only the minimal entitlements required for the transaction.